js/ui/date_box/ui.date_box.mask.js

```diff
@@ -325,10 +325,18 @@ const DateBoxMask = DateBoxBase.inherit({
 
     _prepareRegExpInfo() {
         this._regExpInfo = getRegExpInfo(this._getFormatPattern(), dateLocalization);
-        const regExp = this._regExpInfo.regexp;
-        const flags = regExp.flags;
-        const convertedRegExp = numberLocalization.convertDigits(this._regExpInfo.regexp.source, false);
-        this._regExpInfo.regexp = RegExp(convertedRegExp, flags);
+        const source = this._regExpInfo.regexp.source;
+        const flags = this._regExpInfo.regexp.flags;
+        const quantifiers = new RegExp(/(\{[0-9],?[0-9]?\})/);
```

Comment:
Shoudn't we also take into account the scenario when 2+ digits number is used in quantifier?

For example, current regex will not work for string with `{1, 10}` quantifier